### PR TITLE
CLI cleanup. Do not require client for creating seeds or migrations

### PR DESCRIPTION
**Pull request by kibertoad**
Cleanup and improvements after https://github.com/tgriesser/knex/pull/2884/files
fixes https://github.com/tgriesser/knex/issues/1134

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -55,12 +55,18 @@ function mkConfigObj(opts) {
 function initKnex(env, opts) {
   checkLocalModule(env);
 
-  if (!env.configuration) {
-    if (opts.client) env.configuration = mkConfigObj(opts);
-    else
+  if (!env.pathToKnexFile) {
```
Comment (kibertoad) [inline bin/cli.js]: I've separated `pathToKnexFile` and `configuration` properties to reduce confusion.

File: bin/cli.js [hunk 2]
```diff
@@ -55,12 +55,18 @@ function mkConfigObj(opts) {
...
+    if (opts.client) {
+      env.configuration = mkConfigObj(opts);
+    } else {
       exit(
-        'No knexfile found in this directory. Specify a path with --knexfile'
+        'No knexfile found in this directory. Specify a path with --knexfile or pass --client and --connection params in commandline'
```
Comment (kibertoad) [inline bin/cli.js]: I though this was broken recently, but looks like it was the case for quite a while. Doesn't look like we search for "knexfile in this directory" at all. We probably should.

File: bin/cli.js [hunk 3]
```diff
@@ -165,16 +170,14 @@ function invoke(env) {
       `-x [${filetypes.join('|')}]`,
       'Specify the stub extension (default js)'
     )
-    .action(function(name) {
-      var instance = initKnex(env, commander.opts());
-      var ext = (
-        argv.x ||
-        env.configuration.ext ||
-        env.configuration.split('.').pop()
```
Comment (kibertoad) [inline bin/cli.js]: I have no idea why we were trying to parse knexfile param for extensions, it looks wrong. I'm open to restoring it if someone explains why it was there in the first place.

File: src/util/batchInsert.js [hunk 4]
```diff
@@ -1,84 +1,84 @@
...
-        transaction = tr;
-
-        return this;
-      },
-    }
-  );
-
-  return wrapper;
-}
+import { isNumber, isArray, chunk, flatten, assign } from 'lodash';
```
Comment (kibertoad) [inline src/util/batchInsert.js]: No changes in this file, just reformatted with new Prettier version

File: test/jake/migrate.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env jake
+'use strict';
+/* eslint-disable no-undef */
+
```
Comment (kibertoad) [inline test/jake/migrate.js]: added .js extension for proper syntax colouring

File: test/jake/migrate.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    bin.addListener('stderr', (data) => (stderr += data.toString()));
+    bin.run();
+  });
+}
+
+const taskList = [];
+function test(description, func) {
+  const tmpDirPath = os.tmpdir() + '/knex-test-';
+  let itFails = false;
+  rimrafSync(tmpDirPath);
```
Comment (kibertoad) [inline test/jake/migrate.js]: added temp directory deletion before each test

File: test/jake/migrate.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      )
+    )
+    .then(() =>
+      assertExec(
+        `grep exports.up ${temp}/migrations/*_create_rule_table.js`,
+        'Migration created with boilerplate'
+      )
+    ));
+
+test('Create a migration file without client passed', (temp) =>
```
Comment (kibertoad) [inline test/jake/migrate.js]: new test

**Comment by kibertoad** [comment]
@aurium Could you please review?

File: package.json [hunk 8]
```diff
@@ -86,7 +86,7 @@
     "plaintest": "mocha --exit -t 10000 -b -R spec test/index.js && npm run tape",
     "prepublish": "npm run babel",
     "pre_test": "npm run lint",
-    "bin_test": "jake -f test/jake/*",
+    "bin_test": "jake -f test/jake/migrate.js",
```
Comment (kibertoad) [inline package.json]: at least locally it doesn't execute anything for me unless specific file (or directory with file called "Jakefile") is specified.

Comment (elhigu) [reply package.json]: Maybe that is platform dependent what it does (shell might expand those wildcards which may break it). I remember having to put wildcard notations in quotes at least with mocha to make globs to work. Like `-f "test/jake/*"`

File: src/seed/index.js [hunk 9]
```diff
@@ -124,9 +124,11 @@ Seeder.prototype._waterfallBatch = function(seeds) {
     seed = require(name);
 
     // Run each seed file.
-    current = current.then(() => seed.seed(knex, Promise)).then(function() {
-      log.push(name);
-    });
+    current = current
+      .then(() => seed.seed(knex, Promise))
+      .then(function() {
```
Comment (elhigu) [inline src/seed/index.js]: could be also changed to arrow function

**Comment by kibertoad** [comment]
@elhigu Thanks! It was an insanely bad day of a pretty bad month, really appreciate something positive :D

File: bin/cli.js [hunk 10]
```diff
@@ -37,8 +37,8 @@ function checkLocalModule(env) {
 }
 
 function mkConfigObj(opts) {
-  let envName = opts.env || process.env.NODE_ENV || 'development';
-  let useNullAsDefault = opts.client === 'sqlite3';
+  const envName = opts.env || process.env.NODE_ENV || 'development';
+  const useNullAsDefault = opts.client === 'sqlite3';
```
[the remark this answers is not in the document]
Comment (kibertoad) [reply bin/cli.js]: good point, it does.

File: package.json (hunk 8 above)
Comment (kibertoad) [reply package.json]: You are probably right, I've checked logs for original PR that introduced CLI tests, and they do indeed show
```
> jake -f test/jake/*
☑ Create a migration file
☑ Run migrations
```

So I'll revert this change.

File: bin/cli.js (hunk 3 above)
Comment (kibertoad) [reply bin/cli.js]: But how could it possibly work if `env.configuration` is either path to knexfile.js or resolved configuration object?

Comment (elhigu) [reply bin/cli.js]: buggily 😛

**Pull request merged**